### ISSUE TITLE
Regression in parser of record declarations

### ISSUE THREAD

**Issue by andreasabel** [agda/agda#5236]
Both of these work in 2.6.1 but the second fails on master:
```agda
record Works : Set₁ where
  eta-equality; field
    W : Set

record Test : Set₁ where
  constructor c; field
    F : Set   -- Parse error
```
Discovered this when I tried to showcase some code to students.

**Comment by gallais** [comment]
That's due to #5001 

I am not sure it's possible to have both `constructor` as a layout keyword &
a keyword that can be used inline.

**Comment by andreasabel** [comment]
In this case we have two layout keywords on the same line.  Couldn't we say than the first one isn't treated as introducing layout? 
In general, with several layout keywords on the same line, could we make only the last one count as such?
This is #1145, I suppose.

**Comment by jespercockx** [comment]
This also gives a strange error:
```agda
record R : Set₁ where
  constructor c; eta-equality
  field
    foo : Set
```
Error message:
```
This declaration is illegal in a record before the last field
when scope checking the declaration
  record R where
    constructor
      c
      eta-equality
    field foo : Set
```
This code was accepted on 2.6.1.

**Comment by andreasabel** [comment]
I am intending to fix #1145 in April, which would also fix the present issue.

**Comment by andreasabel** [comment]
> I am intending to fix #1145 in April, which would also fix the present issue.

I was mistaken.  With #1145, we no longer get parse errors, but
```agda
record Test : Set₁ where
  constructor c; field
    F : Set   
```
now is understood as
```agda
record Test : Set₁ where
  constructor 
    c
    field
      F : Set   
```
with respective error messages.

We could however teach Agda to understand constructor blocks in record declarations in a special way, namely, to only take the first name as being inside the block, the rest outside.

Another avenue would be to understand `constructor` as a layout keyword only inside a `interleaved mutual` block.  This would need augmentation of the parser state, but is certainly doable.

**Comment by andreasabel** [comment]
We could store layout-inducing keywords in the stack of layout blocks.  When we encounter `constructor`, we search backwards through the stack.  If we see `mutual` we let `constructor` be a layout keyword, if we see `where` (which is following `record`), we don't.  If the stack is exhausted, we also don't.
